### PR TITLE
fail if YAML doesn't parse

### PR DESCRIPTION
**Pull request by jpsim**
Addresses #310 /cc @NachoSoto

### CODE REVIEW

File: Source/SwiftLintFramework/Models/Configuration.swift [hunk 1]
```diff
@@ -143,10 +141,10 @@ public struct Configuration: Equatable {
                 configPath = fullPath
                 return
             } else {
-                failIfRequired()
+                fail()
```
Comment (NachoSoto) [inline Source/SwiftLintFramework/Models/Configuration.swift]: Makes sense!

**Comment by NachoSoto** [comment]
Awesome :+1:

File: Source/SwiftLintFramework/Models/Configuration.swift (hunk 1 above)
Comment (jpsim) [inline Source/SwiftLintFramework/Models/Configuration.swift]: Ah yeah, that could work too, although the point of the local `fail()` function is to keep the failure-related logic in there anyway, so I'll keep it as is.

**Comment by jpsim** [comment]
Thanks for reviewing!

File: Source/SwiftLintFramework/Models/Configuration.swift (hunk 1 above)
Comment (scottrhoyt) [inline Source/SwiftLintFramework/Models/Configuration.swift]: @NachoSoto @jpsim That is now the [current behavior](https://github.com/realm/SwiftLint/blob/69477dab924ff61854c90903b30090ff74b799de/Source/SwiftLintFramework/Models/Configuration.swift#L107-L109).

**Pull request merged**